numpy/core/tests/test_umath.py

```diff
@@ -1287,6 +1287,71 @@ def test_nan():
 
         assert_raises(TypeError, test_nan)
 
+
+class _CustomFloat(object):
```

Comment:
If there is already an implementation of mocks like these somewhere, I would be happy to get rid of them.

Reply:
Can you use `unittest.mock`?